packages/cli/src/cli/commands/init.test.ts

```diff
@@ -0,0 +1,128 @@
+  await sh(`${PCC} init ${appFolder} --template nextjs --use-pnpm`);
+
+  // Dependencies are installed
+  expect(fs.existsSync(`${appFolder}/node_modules`)).toBe(true);
+  expect(fs.existsSync(`${appFolder}/pnpm-lock.yaml`)).toBe(true);
+
+  // Eslint not initialized
+  // expect(fs.existsSync(`${appFolder}/.eslintrc.json`)).toBe(false);
+
+  // Checking if primary required files for Nextjs starter kit are created.
+  expect(fs.existsSync(`${appFolder}/next.config.js`)).toBe(true);
+  expect(fs.existsSync(`${appFolder}/pages/index.jsx`)).toBe(true);
+
+  // package.json checks
+  expect(fs.existsSync(`${appFolder}/package.json`)).toBe(true);
+  const packageJson = JSON.parse(
+    readFileSync(`${appFolder}/package.json`).toString(),
+  );
+  expect(packageJson.name).toBe(path.parse(appFolder).base);
+
+  // Remove app folder
+  fs.rmSync(appFolder, { recursive: true, force: true });
+});
+
+test("should be able to init starter kit for gatsby template", async () => {
+  const appFolder = tmp.tmpNameSync();
+  await sh(`${PCC} init ${appFolder} --template gatsby --use-pnpm`);
+
+  // Dependencies are installed
+  expect(fs.existsSync(`${appFolder}/node_modules`)).toBe(true);
+  expect(fs.existsSync(`${appFolder}/pnpm-lock.yaml`)).toBe(true);
+
+  // Eslint not initialized
+  expect(fs.existsSync(`${appFolder}/.eslintrc.json`)).toBe(false);
+
+  // Checking if primary required files for Gatsby starter kit are created.
+  expect(fs.existsSync(`${appFolder}/gatsby-browser.js`)).toBe(true);
+  expect(fs.existsSync(`${appFolder}/gatsby-config.js`)).toBe(true);
+  expect(fs.existsSync(`${appFolder}/gatsby-browser.js`)).toBe(true);
+
+  // package.json checks
+  expect(fs.existsSync(`${appFolder}/package.json`)).toBe(true);
+  const packageJson = JSON.parse(
+    readFileSync(`${appFolder}/package.json`).toString(),
+  );
+  expect(packageJson.name).toBe(path.parse(appFolder).base);
+
+  // Remove app folder
+  fs.rmSync(appFolder, { recursive: true, force: true });
+});
+
+test("should be able to init starter kit with eslint and app name", async () => {
+  const appFolder = tmp.tmpNameSync();
+  await sh(
+    `${PCC} init ${appFolder} --appName test_app --template nextjs --noInstall --eslint`,
+  );
+
+  // Eslint is initialized
+  expect(fs.existsSync(`${appFolder}/.eslintrc.json`)).toBe(true);
+
+  // Package.json checks
+  expect(fs.existsSync(`${appFolder}/package.json`)).toBe(true);
+  const packageJson = JSON.parse(
+    readFileSync(`${appFolder}/package.json`).toString(),
+  );
+  expect(packageJson.name).toBe("test_app");
+
+  // Remove app folder
+  fs.rmSync(appFolder, { recursive: true, force: true });
+});
+
+test("should raise error when project directory already exists", async () => {
+  const appFolder = tmp.tmpNameSync();
+  fs.mkdirSync(appFolder);
+
+  let error = 0;
+  try {
+    await sh(
+      `${PCC} init ${appFolder} --appName test_app --template gatsby --noInstall --eslint`,
+    );
+  } catch (err) {
+    error = 1;
```

Comment:
Could this and the other instances be updated to check for the specific error expected to be thrown here? So other regressions don't silently get by the tests

Reply:
Yeah, actually we should. But I am not very sure how to capture stdout from `pcc init` command.
I will give it a try. Any pointers are welcome.

Reply:
We should be able to spy on logger.error then assert it was called in addition to the message it was called with. 
Ref: https://stackoverflow.com/questions/50379916/how-to-mock-test-a-node-js-cli-with-jest

Reply:
Yeah, I had checked this. It doesn't work since we are calling CLI commands using `exec`. `spyOn` doesn't mock method inside child process.

There is one place where I could test JS function instead of CLI command. Updated code to use `spyOn` in it.

Thanks.